src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/PartialViewResultExecutor.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc.ViewEngines;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.OptionsModel;
+
+namespace Microsoft.AspNet.Mvc.ViewFeatures
+{
+    /// <summary>
+    /// Finds and executes an <see cref="IView"/> for a <see cref="PartialViewResult"/>.
+    /// </summary>
+    public class PartialViewResultExecutor : ViewExecutor
+    {
+        /// <summary>
+        /// Creates a new <see cref="ViewResultExecutor"/>.
+        /// </summary>
+        /// <param name="viewOptions">The <see cref="IOptions{MvcViewOptions}"/>.</param>
+        /// <param name="viewEngine">The <see cref="ICompositeViewEngine"/>.</param>
+        /// <param name="telemetry">The <see cref="TelemetrySource"/>.</param>
+        /// <param name="logger">The <see cref="ILogger{PartialViewResultExecutor}"/>.</param>
+        public PartialViewResultExecutor(
+            IOptions<MvcViewOptions> viewOptions,
+            ICompositeViewEngine viewEngine,
+            TelemetrySource telemetry,
+            ILogger<PartialViewResultExecutor> logger)
+            : base(viewOptions, viewEngine, telemetry)
+        {
+            if (logger == null)
+            {
+                throw new ArgumentNullException(nameof(logger));
+            }
+
+            Logger = logger;
+        }
+
+        /// <summary>
+        /// Gets the <see cref="ILogger"/>.
+        /// </summary>
+        protected ILogger Logger { get; }
+
+        /// <summary>
+        /// Attempts to find the <see cref="IView"/> associated with <paramref name="viewResult"/>.
+        /// </summary>
+        /// <param name="actionContext">The <see cref="ActionContext"/> associated with the current request.</param>
+        /// <param name="viewResult">The <see cref="PartialViewResult"/>.</param>
+        /// <returns>A <see cref="ViewEngineResult"/>.</returns>
+        public virtual ViewEngineResult FindView(ActionContext actionContext, PartialViewResult viewResult)
```

Comment:
All of ASP.NET could be one method with callbacks you override. Personally, I like to see the 5-6 lines of code that end up in the result class, and having it all in one method would make this less flexible. What do you think would be better if they were combined?